Modules/Core/src/DataManagement/mitkImage.cpp

```diff
@@ -35,14 +43,184 @@ See LICENSE.txt or http://www.mitk.org for details.
 #define FILL_C_ARRAY( _arr, _size, _value) for(unsigned int i=0u; i<_size; i++) \
 { _arr[i] = _value; }
 
+void mitk::Image::SaveMetaDataDictionaryArraySize(const unsigned int size)
+{
+  m_MetaDataDictionaryArraySize = size;
+}
+
+int indexFromMatrixRow(mitk::Matrix3D matrix, int row)
+{
+  int index = 0;
```

Comment:
int index = 2;
for (int i = 0; i < index; ++i) {
if (!!matrix[row][i]) {
index = i;
break;
}
}